workshops/views.py

```diff
@@ -3154,10 +3147,132 @@ def duplicates(request):
 
     return render(request, 'workshops/duplicates.html', context)
 
+# ------------------------------------------------------------
+# Views for trainees
+
+def _homework2state(homework):
```

Comment:
move to util.py? Or to models.py - depending on the function's purpose (missing docstring)